lisa/util/package.py

```diff
@@ -24,6 +26,19 @@
 
 """
 
+LINUX_ONLY_MODULES = {
+    "sut_orchestrator/qemu",
```

Comment:
It's good to add comments to explain why it's excluded. I know the libvirt doesn't support Windows, but we can let others know.

Reply:
How about use the regular expression? It's powerful and makes the following logic simpler. I refer to use `re.compile` here to add compiled patterns for speeding up.

Reply:
I decided to use the .gitignore format.

Reply:
Is there a module supporting the .gitignore format? The LISA use the "re" module a lot, to keep consistent, I prefer to use regular expression.

Reply:
Yeah, I found the `gitignore_parser` package. While it isn't exactly a well known package, the codebase is small enough that I was able to quickly scan through it and found nothing of concern.

Reply:
Hmm, I am thinking I should probably design something that also works for extensions, including for tests that live outside of this repo.

Reply:
If it needs in extensions, it can be added in the extension's schema. It's not a high priority feature, because it can be avoided by create multiple extension folders.

Reply:
I tried something different: Instead of dynamically loading all the missing modules, I just manually `import` all the missing modules. This allows for the more pythonic pattern of adding `if platform.system() == "Linux"` style checks over the `import` statements.

I left dynamic package loading in place for extensions. Though I did add the ability to only load a single python file. So, if an extension has platform specific imports, then they can create a `module.py` file that handles the imports correctly and then only load the `module,py` file for the extension.

Thoughts on this approach?

Reply:
Let's discuss about extension only here. Actually, I have similar requirements to exclude some of files in an extension folder. But it should be dynamically, decide by runbook, not in code. For example, there is a NSG rule modification extension, it should apply to MS subscriptions only.
I think we can add something into extension's schema and implement like below.

```diff
class Extension:
    path: str
    name: Optional[str] = None
+   excludes: Optional[List[str]] = None
```

Reply:
It is fairly common for test execution frameworks to have a concept of tagging (e.g. MSTest's [TestCategory](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.testcategoryattribute?view=visualstudiosdk-2022)). It is possible your scenario might be better served by such a feature.

Reply:
There are some tags like Platform already. It uses to limit test case on supported platforms. The NSG scenario is not related to test case actually, it's related to environments. so it's not fit to be a tag on test case, it should be configurable in runbook. Anyway, I will implement it later. Let me know, if it needs higher priority.

Reply:
If you were to add a generic tagging feature, then it would probably make sense to allow tag filters to be specified in the runbook.

Reply:
There is tag in test suite metadata. But it doesn't work for module loading, the module loading happens very early. The tag is used to categorize tests, not for other purposes. We can discuss about the scenarios to see what's needed here.

Reply:
Is your scenario that you need to apply special network firewall rules for VMs when testing under a MS subscription?

Reply:
That's same to all MS subscriptions. If a machine, which is not a part of corp network, it needs an extra NSG rule to SSH VM in MS subscription.